v2/tools/generator/internal/config/type_configuration.go

```diff
@@ -54,11 +55,18 @@ const (
 	isResourceTag               = "$isResource"               // Boolean specifying whether a particular type is a resource or not.
 	manualConfigsTag            = "$manualConfigs"            // A set of strings specifying which config map fields should be generated (to be filled out by resource extension)
 	nameInNextVersionTag        = "$nameInNextVersion"        // String specifying a type or property name change in the next version
+	operatorSpecPropertiesTag   = "$operatorSpecProperties"   // A set of additional properties to inject into the operatorSpec of a resource
 	renameTo                    = "$renameTo"                 // String specifying the new name of a type
 	resourceEmbeddedInParentTag = "$resourceEmbeddedInParent" // String specifying resource name of parent
 	supportedFromTag            = "$supportedFrom"            // Label specifying the first ASO release supporting the resource
 )
 
+type OperatorSpecPropertyConfiguration struct {
+	Name        string `yaml:"name"`        // Name of the new property
```

Comment:
minor, style, feel free to ignore: Could promote the comments to actual godoc comments:
```suggestion
        // Name of the new property
	Name        string `yaml:"name"` 
```

Reply:
Could also spell out more explicitly on the `Type` how it only supports primitive types.

Reply:
What are you thinking of? The current comment seems pretty explicit to me:
``` go
// Primitive type of the new property  (e.g. string, int, etc.)
```

Reply:
It sure does. Must have been the end of the day lol